bolt-simple-taproot.md

```diff
@@ -0,0 +1,1320 @@
+incremental updates will be easier to implement and review, and may also
+hasten the timeline to deploy initial taproot based channels.
+
+In this document, we start by revising the most fundamental component of a
+Lightning Channel: the funding output. By first porting the funding output to
+Segwit V1 (P2TR) `musig2` based output, we're able to re-anchor all channels in
+the network with the help of dynamic commitments. Once those channels are
+re-anchored, dynamic commitments allows developers to ship incremental changes
+to the commitment transaction, HTLC structure, and overall channel structure --
+all without additional on-chain transactions.
+
+By restricting the surface area of the _initial_ update, we aim to expedite the
+initial roll-out while also providing a solid base for future updates without
+blocking off any interesting design paths. Implementing and shipping a
+constrained update also gives implementers an opportunity to get more familiar
+with the new tooling and capabilities, before tackling more advanced protocol
+designs.
+
+## Preliminaries
+
+In this section we lay out some preliminary concepts, protocol flows, and
+notation that'll be used later in the core channel specification.
+
+### Pay-To-Taproot-Outputs
+
+The Taproot soft fork package (BIPs
+[340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki),
+[341](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki), and
+[342](https://github.com/bitcoin/bips/blob/master/bip-0342.mediawiki))
+introduced a new
+[Segwit](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki)
+witness version: v1. This new witness version utilizes a new standardized
+public key script template:
+```
+OP_1 <taproot_output_key>
+```
+
+The `taproot_output_key` is a 32-byte x-only `secp256k1` public key, with all
+digital signatures based off of the schnorr signature scheme described in [BIP
+340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki).
+
+A `taproot_output_key` commits to an internal key, and optional script root via
+the following mapping:
+```
+taproot_output_key = taproot_internal_key + tagged_hash("TapTweak", taproot_internal_key || script_root)*G
+```
+
+It's important to note that while `taproot_output_key` is serialized as a
+32-byte public key, in order to properly spend the script path, the parity
+(even or odd) of the output public key must be remembered.
+
+The `taproot_internal_key` is also a BIP 340 public key, ideally derived anew
+for each output. The `tagged_hash` scheme is described in BIP 340, we briefly
+define the function as:
+```
+tagged_hash(tag, msg) = SHA256(SHA256(tag) || SHA256(tag) || msg)
+```
+
+#### Tapscript Tree Semantics
+
+The `script_root` is the root of a Tapscript tree as defined in BIP 341. A tree
+is composed of `tap_leaves` and `tap_branches`.
+
+A `tap_leaf` is a two tuple of (`leaf_version`, `leaf_script`). A `tap_leaf` is serialized as:
```

Comment:
i wonder if the correctness of the `leaf_version` (i.e == `0xc0` or == `0xc1`) is verified once the funding output has hit the chain (no `option_zero_conf` mode) to be sure the `script_root` built from counterparty contributions  is the correct one committed on-chain.

from looking on `channel_ready` recipient, there is no say that the `script_root` must be verified. it might be elsewhere in this document. e.g for offered HTLCs there is a split of the timeout and success paths in individual script leaves.